airbyte-integrations/connectors/source-weatherstack/source_weatherstack/source.py

```diff
@@ -2,332 +2,26 @@
-
-    def path(
-        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> str:
-        return "autocomplete"
-
-    def request_params(
-        self,
-        stream_state: Mapping[str, Any],
-        stream_slice: Mapping[str, Any] = None,
-        next_page_token: Mapping[str, Any] = None,
-    ) -> MutableMapping[str, Any]:
-        # The api requires that we include api_key as a query param so we do that in this method
-        return {"access_key": self.access_key, "query": self.query}
-
-    def backoff_time(self, response: requests.Response) -> Optional[float]:
-        delay_time = response.headers.get("Retry-After")
-        if delay_time:
-            return int(delay_time)
-
-    def parse_response(
-        self,
-        response: requests.Response,
-        stream_state: Mapping[str, Any],
-        stream_slice: Mapping[str, Any] = None,
-        next_page_token: Mapping[str, Any] = None,
-    ) -> Iterable[Mapping]:
-        # The response is a simple JSON whose schema matches our stream's schema exactly,
-        # so we just return a list containing the response
-        return [response.json()]
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        # The API does not offer pagination,
-        # so we return None to indicate there are no more pages in the response
-        return None
-
-
-# Source
-class SourceWeatherstack(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
-
-        try:
-            query = config["query"]
-            access_key = config["access_key"]
-
-            response = requests.get(f"{url_base}/current?access_key={access_key}&query={query}")
-            response = response.text
-
-            if response.find('"success": false') != -1:
-                return False, "Check Query and Access Key"
-            else:
-                return True, None
-        except requests.exceptions.RequestException as e:
-            return False, repr(e)
+# Declarative Source
+class SourceWeatherstack(YamlDeclarativeSource):
+    def __init__(self):
+        super().__init__(**{"path_to_yaml": "manifest.yaml"})
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
-        auth = None
-        streams = [
-            CurrentWeather(authenticator=auth, config=config),
-            Forecast(authenticator=auth, config=config),
-        ]
-
-        # Historical stream is only supported by paid accounts
-        if config["is_paid_account"] is not False:
-            streams.append(LocationLookup(authenticator=auth, config=config))
-            streams.append(Historical(authenticator=auth, config=config))
-
-        return streams
+        streams = super().streams(config)
+        if not config.get("is_paid_account"):
+            return [stream for stream in streams if stream.name not in ["location_lookup", "historical"]]
```

Comment:
Instead of hiding streams altogether, let's keep the streams, but setup a custom error handler. Use error handler to intercept an error code (it would likely be something like HTTP 403 with a specific error message). When you get this error, instead of raising, swallow them.

Reply:
Got it @natikgadzhi 

But do I still need to remove the is_paid_account config input? That would introduce a breaking change, i suppose

Reply:
Actually, the api doesn't raise an error, so there is technically nothing to handle if the request is bad.

The status code is always 200 regardless and the body of the response is what determines if it's an error or not.

Reply:
Yeah if there are just no errors but empty response, this is fine. 

I wonder if we should keep the config option or drop it. Dropping means breaking change, but if it's not used at all, we should drop it.